pkg/commands/resolver.go

```diff
@@ -61,20 +61,22 @@ func gobuildOptions(bo *options.BuildOptions) ([]build.Option, error) {
 		return nil, err
 	}
 
-	if bo.Platform == "" {
-		bo.Platform = "linux/amd64"
+	if len(bo.Platforms) == 0 {
+		envPlatform := "linux/amd64"
 
 		goos, goarch, goarm := os.Getenv("GOOS"), os.Getenv("GOARCH"), os.Getenv("GOARM")
 
 		// Default to linux/amd64 unless GOOS and GOARCH are set.
 		if goos != "" && goarch != "" {
-			bo.Platform = path.Join(goos, goarch)
+			envPlatform = path.Join(goos, goarch)
 		}
 
 		// Use GOARM for variant if it's set and GOARCH is arm.
 		if strings.Contains(goarch, "arm") && goarm != "" {
-			bo.Platform = path.Join(bo.Platform, "v"+goarm)
+			envPlatform = path.Join(envPlatform, "v"+goarm)
 		}
+
+		bo.Platforms = append(bo.Platforms, envPlatform)
```

Comment:
We know `bo.Platforms` is empty at this point, I think it's a bit clearer to say `bo.Platforms = []string{envPlatform}`

Reply:
hmm yeah I think I thought too much about allocations when I wrote it 😆